pkg/sql/logictest/testdata/logic_test/fk

```diff
@@ -2657,7 +2657,9 @@ user testuser
 
 # Run an INSERT which needs to check existence of the row. If we try to scan
 # the entire row, this blocks on the other transaction. We should only be
-# scanning the primary column family.
+# scanning the primary column family. A critical reason why this works is
+# because column b is NOT NULL, so the UPDATE statement does not acquire
+# FOR UPDATE locks on the primary column family.
```

Comment:
If we acquired ranged locks on reads, would we just get a lock on `b` even if it's `NULL`, or would we still want to lock the PK?